### DISCUSSION TITLE
✅ Listening to a configurable hostname

### DISCUSSION THREAD

**Discussion by huyhoang160593** [RobinTail/express-zod-api#1268] [Ideas]
I'm tried to find the config that can let me to use different hostname (e.g 0.0.0.0) than the localhost but can't find any in the documentation, shouldn't we have that anywhere or I have to use a custom express app in order to achieve that?

**Comment by RobinTail** [comment]
Hello, @huyhoang160593 .

When using `createServer` method, the created server will start listening on the specified port or UNIX socket of _unspecified_ host.

> If host is omitted, the server will accept connections on the [unspecified IPv6 address](https://en.wikipedia.org/wiki/IPv6_address#Unspecified_address) (::) when IPv6 is available, or the [unspecified IPv4 address](https://en.wikipedia.org/wiki/0.0.0.0) (0.0.0.0) otherwise.
>
> https://nodejs.org/api/net.html#serverlistenport-host-backlog-callback

So, currently host is not configurable for `createServer` and only achievable using `attachRouting` method on a custom express app.

However, it should be easy to implement this feature by allowing `listen` config property to accept object options.

**Comment by RobinTail** [comment]
@huyhoang160593 ,
the feature is being implemented in the following PR: #1270

**Comment by RobinTail** [comment]
🚀  The feature released in version 14.1.0, @huyhoang160593

**Reply by huyhoang160593** [reply]
Thank you so much for the fast response and implementation, I will update the library right away and check them out
